api/oidc_test.go

```diff
@@ -0,0 +1,143 @@
+package api_test
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/buildkite/agent/v3/api"
+	"github.com/buildkite/agent/v3/logger"
+	"github.com/google/go-cmp/cmp"
+)
+
+func newOidcTokenServer(
+	t *testing.T,
```

Comment:
It looks like the `t` argument is only used for calling `t.Helper()`. If so there's likely no point in passing in the `t` at all (compare with https://github.com/golang/go/wiki/TestComments#mark-test-helpers - the `readFile` example calls `t.Fatal` but `t.Helper` ensures the line in the test itself is identified as the problem.)

Reply:
The `t` is used. `t.Error` is called when the body of the request does not match what is expected.

Reply:
Ah, I missed that when I read too quickly. It does raise a question:

_Should it_ call `t.Error`? Without it, the test should still fail when the request mismatches and the test server returns a 400. With it, the test server is doing more than merely faking the real server?

No need to block on this one.

Reply:
I think I have a solution for this. I'll make a new PR for it though.